benchmark/timings-geometry.cpp

```diff
@@ -0,0 +1,220 @@
+  se3::Model romeo_model = romeo.first;
+  se3::GeometryModel romeo_model_geom = romeo.second;
+  Data romeo_data(romeo_model);
+  GeometryData romeo_data_geom(romeo_data, romeo_model_geom);
+
+
+  std::vector<VectorXd> qs_romeo     (NBT);
+  std::vector<VectorXd> qdots_romeo  (NBT);
+  std::vector<VectorXd> qddots_romeo (NBT);
+  for(size_t i=0;i<NBT;++i)
+  {
+    qs_romeo[i]     = Eigen::VectorXd::Random(romeo_model.nq);
+    qs_romeo[i].segment<4>(3) /= qs_romeo[i].segment<4>(3).norm();
+    qdots_romeo[i]  = Eigen::VectorXd::Random(romeo_model.nv);
+    qddots_romeo[i] = Eigen::VectorXd::Random(romeo_model.nv);
+  }
+
+  timer.tic();
+  SMOOTH(NBT)
+  {
+    geometry(romeo_model,romeo_data,qs_romeo[_smooth]);
+  }
+  double geom_time = timer.toc(StackTicToc::US)/NBT;
+
+  timer.tic();
+  SMOOTH(NBT)
+  {
+    updateCollisionGeometry<true>(romeo_model,romeo_data,romeo_model_geom,romeo_data_geom,qs_romeo[_smooth]);
+  }
+  double update_col_time = timer.toc(StackTicToc::US)/NBT - geom_time;
+  std::cout << "Update Collision Geometry < false > = \t" << update_col_time << " " << StackTicToc::unitName(StackTicToc::US) << std::endl;
+
+  timer.tic();
+  SMOOTH(NBT)
+  {
+    updateCollisionGeometry<true>(romeo_model,romeo_data,romeo_model_geom,romeo_data_geom,qs_romeo[_smooth]);
+    for (std::vector<se3::GeometryData::CollisionPair_t>::iterator it = romeo_data_geom.collision_pairs.begin(); it != romeo_data_geom.collision_pairs.end(); ++it)
+    {
+      romeo_data_geom.collide(it->first, it->second);
+    }
+  }
+  double collideTime = timer.toc(StackTicToc::US)/NBT - (update_col_time + geom_time);
+  std::cout << "Collision test between two geometry objects (mean time) = \t" << collideTime / romeo_data_geom.nCollisionPairs
+            << StackTicToc::unitName(StackTicToc::US) << std::endl;
+
+
+  timer.tic();
+  SMOOTH(NBT)
+  {
+    updateCollisionGeometry<true>(romeo_model,romeo_data,romeo_model_geom,romeo_data_geom,qs_romeo[_smooth]);
+    romeo_data_geom.isColliding();
+  }
+  double is_colliding_time = timer.toc(StackTicToc::US)/NBT - (update_col_time + geom_time);
+  std::cout << "Collision Test : robot in collision? = \t" << is_colliding_time
+            << StackTicToc::unitName(StackTicToc::US) << std::endl;
+
+
+  timer.tic();
+  SMOOTH(1000)
+  {
+    updateCollisionGeometry<true>(romeo_model,romeo_data,romeo_model_geom,romeo_data_geom,qs_romeo[_smooth]);
+    romeo_data_geom.computeDistances();
+  }
+  double computeDistancesTime = timer.toc(StackTicToc::US)/1000 - (update_col_time + geom_time);
+  std::cout << "Compute distance between two geometry objects (mean time) = \t" << computeDistancesTime / romeo_data_geom.nCollisionPairs
+            << " " << StackTicToc::unitName(StackTicToc::US) << " " << romeo_data_geom.nCollisionPairs << " col pairs" << std::endl;
+
+
+
+#ifdef WITH_HPP_MODEL_URDF
+
+  std::string filenameHRP2 = PINOCCHIO_SOURCE_DIR"/models/hrp2_14_reduced.urdf";
```

Comment:
I would agree that the access to the model is restricted. It usually better to have separate repository and codes because of the psychological aspects. For some reasons a lot of people seems to overlook the preprocessing variables. It is probably much more related to the fact that it is cluttering the code.